prow/spyglass/spyglass.go

```diff
@@ -148,3 +149,89 @@ func (s *Spyglass) JobPath(src string) (string, error) {
 		return "", fmt.Errorf("unrecognized key type for src: %v", src)
 	}
 }
+
+// RunPath returns the path to the GCS directory for the job run specified in src.
+func (s *Spyglass) RunPath(src string) (string, error) {
+	src = strings.TrimSuffix(src, "/")
+	keyType, key, err := splitSrc(src)
+	if err != nil {
+		return "", fmt.Errorf("error parsing src: %v", src)
+	}
+	switch keyType {
+	case gcsKeyType:
+		return key, nil
+	case prowKeyType:
+		return s.prowToGCS(key)
+	default:
+		return "", fmt.Errorf("unrecognized key type for src: %v", src)
+	}
+}
+
+// RunToPR returns the (org, repo, pr#) tuple referenced by the provided src.
+// Returns an error if src does not reference a job with an associated PR.
+func (s *Spyglass) RunToPR(src string) (string, string, int, error) {
+	src = strings.TrimSuffix(src, "/")
+	keyType, key, err := splitSrc(src)
+	if err != nil {
+		return "", "", 0, fmt.Errorf("error parsing src: %v", src)
+	}
+	split := strings.Split(key, "/")
+	if len(split) < 2 {
+		return "", "", 0, fmt.Errorf("expected more URL components in %q", src)
+	}
+	switch keyType {
+	case gcsKeyType:
+		// In theory, we could derive this information without trying to parse the URL by instead fetching the
+		// data from uploaded artifacts. In practice, that would not be a great solution: it would require us
+		// to try pulling two different metadata files (one for bootstrap and one for podutils), then parse them
+		// in unintended ways to infer the original PR. Aside from this being some work to do, it's also slow: we would
+		// like to be able to always answer this request without needing to call out to GCS.
+		logType := split[1]
+		if logType == "logs" {
```

Comment:
Could we use constants and share it with the code that does `gcsupload`?

Reply:
Done.